cpp/ql/src/Security/CWE/CWE-676/DangerousUseOfGets.ql

```diff
@@ -0,0 +1,18 @@
+/**
+ * @name Use of dangerous function 'gets'
+ * @description The standard library 'gets' function is dangerous and should not be used.
+ * @kind problem
+ * @problem.severity error
+ * @precision very-high
+ * @id cpp/potentially-dangerous-function
+ * @tags reliability
+ *       security
+ *       external/cwe/cwe-242
```

Comment:
Shouldn't this be 676, not 242?

Reply:
[CWE-242](https://cwe.mitre.org/data/definitions/242.html) ('Use of Inherently Dangerous Function') is probably the more accurate CWE in this case and the one the CWE examples and samate tests associated with this query are under; where as [CWE-676](https://cwe.mitre.org/data/definitions/676.html) 'Use of Potentially Dangerous Function' is a more general CWE that we have several queries under, hence the directory name.  According to https://wiki.semmle.com/display/IN/Modelling+CWEs+at+Semmle "The rule we currently follow is to add @cwe tags for all of the most specific (using the parent/child relationship in View 1000) Weakness Bases or Weakness Classes which apply to a query." so I *think* this should be tagged CWE-242.  I'm happy to be corrected on this though.